index.js

```diff
@@ -576,6 +576,18 @@ class Hyperdrive extends Nanoresource {
     })
   }
 
+  readFileSync(name, opts){
+    if (typeof opts === 'string') opts = { encoding: opts }
+    if (!opts) opts = {}
+    name = fixName(name)
+    collect(this.createReadStream(name, opts), function (err, bufs) {
+      if (err) throw err
+      let buf = bufs.length === 1 ? bufs[0] : Buffer.concat(bufs)
+      return opts.encoding && opts.encoding !== 'binary' ? buf.toString(opts.encoding) : buf
+    })
+
+  }
```

Comment:
Here, for example, the function main logic is similar to the async method, but error handling and treatment of the results are treated differently.

Reply:
Your return above just returns inside the callback function of collect so if you tried running this method it would always immediately return undefined

Reply:
Yes, I just saw it.